### PR TITLE
[DO-NOT-MERGE][KYUUBI #2619] Add profile spark-3.3

### LINKED ISSUE #2619: [Subtask] Add profile spark-3.3
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues) and found no similar issues.

### Describe the subtask

Add profile spark-3.3

### Parent issue

#2608 

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by cfmcgrady**
### _Why are the changes needed?_

close #2619

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/kyuubi/sql/watchdog/MaxPartitionStrategy.scala [hunk 1]
```diff
@@ -97,7 +97,7 @@ case class MaxPartitionStrategy(session: SparkSession)
                 _),
               _,
               _,
-              _)) if fsRelation.partitionSchemaOption.isDefined =>
+              _)) if fsRelation.partitionSchema.nonEmpty =>
```
Comment (cfmcgrady) [inline extensions/spark/kyuubi-extension-spark-common/src/main/scala/org/apache/kyuubi/sql/watchdog/MaxPartitionStrategy.scala:100]: `HadoopFsRelation.partitionSchemaOption` was removed by https://github.com/apache/spark/pull/34582

File: extensions/spark/kyuubi-extension-spark-3-3/src/main/scala/org/apache/kyuubi/sql/KyuubiSparkSQLParser.scala [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  override def parseTableSchema(sqlText: String): StructType = {
+    delegate.parseTableSchema(sqlText)
+  }
+
+  override def parseDataType(sqlText: String): DataType = {
+    delegate.parseDataType(sqlText)
+  }
+
+  override def parseQuery(sqlText: String): LogicalPlan = {
```
Comment (cfmcgrady) [inline extensions/spark/kyuubi-extension-spark-3-3/src/main/scala/org/apache/kyuubi/sql/KyuubiSparkSQLParser.scala:104]: the new abstract function `parseQuery` was added by https://github.com/apache/spark/pull/34543

**Comment by cfmcgrady** [comment]
GA passed.

1. Ignore flaky test `JpsApplicationOperation with spark local mode`. https://github.com/apache/incubator-kyuubi/issues/2642
2. Disable module `kyuubi-spark-authz` for spark-3.3, we can add support in a separate PR.
3. add maven profile `spark-3.3`, and also add the GitHub action job.
4. all existing spark extensions have passed CI.

cc @pan3793 @ulysses-you @turboFei

**Comment by ulysses-you** [comment]
> Disable module kyuubi-spark-authz for spark-3.3, we can add support in a separate PR.

cc @yaooqinn

**Comment by ulysses-you** [comment]
@cfmcgrady  thank you for the investigation !

**Comment by yaooqinn** [comment]
> Disable module kyuubi-spark-authz for spark-3.3, we can add support in a separate PR.

it breaks ?

**Comment by cfmcgrady** [comment]
> > Disable module kyuubi-spark-authz for spark-3.3, we can add support in a separate PR.
> 
> it breaks ?

yes, it also happen in the master branch. https://github.com/apache/incubator-kyuubi/runs/6416448132?check_suite_focus=true

**Comment by yaooqinn** [comment]
shall we fix the failure first? spark release process for major one take quite a long time

**Comment by cfmcgrady** [comment]
> shall we fix the failure first? spark release process for major one take quite a long time



> shall we fix the failure first? spark release process for major one take quite a long time

Ignore the flaky test to pass CI, to see whether we have other compatibility issues. I will enable the unit test after we fix the issue.

**Comment by yaooqinn** [comment]
seems we are not discussing the same thing，I only mean removing authz module is unacceptable as we have PRs for it now.

**Comment by cfmcgrady** [comment]
> seems we are not discussing the same thing，I only mean removing authz module is unacceptable as we have PRs for it now.

Oops, I misread before.

we have 31 failed unit tests, I think we can raise a separate pr to fix it based on this.

FYI: https://github.com/apache/incubator-kyuubi/runs/6418487737?check_suite_focus=true

**Pull request closed without merging**